implementations/rust/router/src/router.rs

```diff
@@ -0,0 +1,168 @@
+#![allow(unused)]
+pub mod router {
+    use ockam_message::message::*;
+    use std::collections::HashMap;
+    use std::io::{Error, ErrorKind, Read, Write};
+    use std::net::UdpSocket;
+    use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+    use std::ops::Add;
+    use std::sync::{Arc, Mutex};
+    use std::thread;
+
+    pub struct UdpAddressHandler {
+        pub socket: UdpSocket,
+    }
+
+    impl HandleMessage for UdpAddressHandler {
+        fn handle_message(&self, mut m: Box<Message>) -> Result<(), std::io::Error> {
+            println!("In UdpAddressHandler!");
+            Ok(())
+        }
+    }
+
+    pub trait HandleMessage {
```

Comment:
I'm call this `MessageHandler` in elixir, let's use that same name here.

Reply:
ok